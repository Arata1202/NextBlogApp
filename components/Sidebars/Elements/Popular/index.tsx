'use client';
import { useTheme } from 'next-themes';
import React from 'react';
import Image from 'next/image';
import styles from './index.module.css';
import { BoltIcon } from '@heroicons/react/24/solid';
import { PopularPost } from '@/section/dummy';

export default function Popular() {
  const { theme } = useTheme();
  return (
    <div className={`pt-8 px-4 border py-5 mt-5 ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}>
      <h1 className={`${styles.profile} text-2xl text-center font-semibold flex justify-center`}>
        <BoltIcon className="h-8 w-8 mr-2" aria-hidden="true" />
        おすすめの投稿
      </h1>
      {PopularPost.map((item, index) => (
        <ol
          key={item.postName}
          className={`ArticleListItem_list border mt-5 p-2 shadow-lg hover:shadow-xl transition-shadow duration-200 transform hover:-translate-y-1 ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}
        >
          <li>
            <a
              href={item.postHref}
              className="sidebarPopularPost"
              onClick={(event: React.MouseEvent<HTMLAnchorElement>) => {
                event.preventDefault();
                window.location.href = item.postHref;
              }}
            >
              <Image
                src={item.imageHref}
                alt={item.imageAlt}
                className="ArticleListItem_image"
                width="600"
                height="300"
                key={index}
              />
              <div>
                <div className="ArticleListItem_title font-bold">{item.postName}</div>
              </div>
            </a>
          </li>
        </ol>
      ))}
    </div>
  );
}
