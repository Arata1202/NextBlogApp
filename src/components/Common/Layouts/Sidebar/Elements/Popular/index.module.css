@media (max-width: 1023px) and (min-width: 641px) {
  .link {
    display: flex;
    gap: 10px;
  }

  .image {
    max-width: 300px;
  }

  .title {
    font-size: 1.3rem;
  }
}
