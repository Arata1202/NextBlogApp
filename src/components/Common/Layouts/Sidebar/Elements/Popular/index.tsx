'use client';

import { useTheme } from 'next-themes';
import { BoltIcon } from '@heroicons/react/24/solid';
import styles from './index.module.css';
import { PopularPost } from '@/constants/Blog';

export default function Popular() {
  const { theme } = useTheme();

  return (
    <>
      <div
        className={`pt-8 px-4 border py-5 mt-5 ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}
      >
        <div className={`text-2xl text-center font-semibold flex justify-center`}>
          <BoltIcon className="h-8 w-8 mr-2" />
          おすすめの投稿
        </div>
        {PopularPost.map((item, index) => (
          <ul
            key={item.postName}
            className={`border mt-5 p-2 shadow-lg hover:shadow-xl transition-shadow duration-200 transform hover:-translate-y-1 ${theme === 'dark' ? 'DarkTheme' : 'LightTheme'}`}
          >
            <li>
              <a href={item.postHref} className={styles.link}>
                <img
                  key={index}
                  src={item.imageHref}
                  alt={item.imageAlt}
                  className={styles.image}
                />
                <div className={`${styles.title} font-bold`}>{item.postName}</div>
              </a>
            </li>
          </ul>
        ))}
      </div>
    </>
  );
}
