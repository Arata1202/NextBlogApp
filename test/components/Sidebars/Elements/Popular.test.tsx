import { render } from '@testing-library/react';
import Popular from '@/components/Sidebars/Elements/Popular';

describe('Popular', () => {
  test('スナップショット（Popular）', () => {
    const { asFragment } = render(<Popular />);
    expect(asFragment()).toMatchSnapshot();
  });
});
