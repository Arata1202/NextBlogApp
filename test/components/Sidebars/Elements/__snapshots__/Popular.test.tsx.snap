// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Popular スナップショット（Popular） 1`] = `
<DocumentFragment>
  <div
    class="pt-8 px-4 border py-5 mt-5 LightTheme"
  >
    <h1
      class="profile text-2xl text-center font-semibold flex justify-center"
    >
      <svg
        aria-hidden="true"
        class="h-8 w-8 mr-2"
        data-slot="icon"
        fill="currentColor"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M14.615 1.595a.75.75 0 0 1 .359.852L12.982 9.75h7.268a.75.75 0 0 1 .548 1.262l-10.5 11.25a.75.75 0 0 1-1.272-.71l1.992-7.302H3.75a.75.75 0 0 1-.548-1.262l10.5-11.25a.75.75 0 0 1 .913-.143Z"
          fill-rule="evenodd"
        />
      </svg>
      人気の投稿
    </h1>
    <ol
      class="ArticleListItem_list border mt-5 p-2 shadow-lg hover:shadow-xl transition-shadow duration-200 transform hover:-translate-y-1 LightTheme"
    >
      <li>
        <a
          class="sidebarPopularPost"
          href="/articles/nyuto-hotspring-tsurunoyu-reservation-access"
        >
          <img
            alt="サムネイル"
            class="ArticleListItem_image"
            data-nimg="1"
            decoding="async"
            height="300"
            loading="lazy"
            src="/_next/image?url=%2Fimages%2Fpost%2F1.webp&w=1200&q=75"
            srcset="/_next/image?url=%2Fimages%2Fpost%2F1.webp&w=640&q=75 1x, /_next/image?url=%2Fimages%2Fpost%2F1.webp&w=1200&q=75 2x"
            style="color: transparent;"
            width="600"
          />
          <div>
            <div
              class="ArticleListItem_title font-bold"
            >
              【乳頭温泉郷】鶴の湯に宿泊！予約方法やアクセスについて解説
            </div>
          </div>
        </a>
      </li>
    </ol>
    <ol
      class="ArticleListItem_list border mt-5 p-2 shadow-lg hover:shadow-xl transition-shadow duration-200 transform hover:-translate-y-1 LightTheme"
    >
      <li>
        <a
          class="sidebarPopularPost"
          href="/articles/university-student-real-personal-effects"
        >
          <img
            alt="サムネイル"
            class="ArticleListItem_image"
            data-nimg="1"
            decoding="async"
            height="300"
            loading="lazy"
            src="/_next/image?url=%2Fimages%2Fpost%2F2.webp&w=1200&q=75"
            srcset="/_next/image?url=%2Fimages%2Fpost%2F2.webp&w=640&q=75 1x, /_next/image?url=%2Fimages%2Fpost%2F2.webp&w=1200&q=75 2x"
            style="color: transparent;"
            width="600"
          />
          <div>
            <div
              class="ArticleListItem_title font-bold"
            >
              【文系】大学生必見！大学でのリアルな持ち物を大公開【かばんの中身】
            </div>
          </div>
        </a>
      </li>
    </ol>
    <ol
      class="ArticleListItem_list border mt-5 p-2 shadow-lg hover:shadow-xl transition-shadow duration-200 transform hover:-translate-y-1 LightTheme"
    >
      <li>
        <a
          class="sidebarPopularPost"
          href="/articles/study-php-recommend-method"
        >
          <img
            alt="サムネイル"
            class="ArticleListItem_image"
            data-nimg="1"
            decoding="async"
            height="300"
            loading="lazy"
            src="/_next/image?url=%2Fimages%2Fpost%2F3.webp&w=1200&q=75"
            srcset="/_next/image?url=%2Fimages%2Fpost%2F3.webp&w=640&q=75 1x, /_next/image?url=%2Fimages%2Fpost%2F3.webp&w=1200&q=75 2x"
            style="color: transparent;"
            width="600"
          />
          <div>
            <div
              class="ArticleListItem_title font-bold"
            >
              【勉強法】１か月で習得！PHP学習のおすすめロードマップを紹介【プログラミング】
            </div>
          </div>
        </a>
      </li>
    </ol>
  </div>
</DocumentFragment>
`;
